src/App.tsx

```diff
@@ -15,36 +16,98 @@
+  RESET = 'Reset',
+}
+
+const preparedGoods = [...goodsFromServer];
+
+const getSortedGoods = (order: SortType, isReversed: boolean): string[] => {
+  const sortedGoods = [...preparedGoods].sort((a, b) => {
+    switch (order) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.RESET);
+
+  const handleSortOrderChange = (order: SortType) => {
+    if (order === SortType.RESET) {
+      setIsReversed(false);
+    }
+
+    setSortOrder(order);
+  };
+
+  const handleReverseGoods = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const goods = getSortedGoods(sortOrder, isReversed);
+  const isResetVisible = sortOrder !== SortType.RESET || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortOrder !== SortType.ALPHABET,
+          })}
+          onClick={() => handleSortOrderChange(SortType.ALPHABET)}
+        >
+          {SortType.ALPHABET}
         </button>
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortOrder !== SortType.LENGTH,
+          })}
+          onClick={() => handleSortOrderChange(SortType.LENGTH)}
+        >
+          {SortType.LENGTH}
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={handleReverseGoods}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isResetVisible && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={() => handleSortOrderChange(SortType.RESET)}
+          >
+            {SortType.RESET}
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {goods.map(good => (
+          <li key={good} data-cy="Good">
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, make sure to add a `key` prop that is unique for each element. Using the `good` string itself as a key might not be unique if the list contains duplicate items. It's better to use an identifier that is guaranteed to be unique, such as an ID.